warehouse/query-core/src/main/java/datawave/query/language/parser/jexl/JexlNodeSet.java

```diff
@@ -93,84 +103,45 @@ public boolean contains(Object o) {
     
     @Override
     public boolean add(JexlNode node) {
-        if (isDelayed(node)) {
-            if (log.isTraceEnabled()) {
-                log.trace("Trying to add a delayed node: " + nodeToKey(node));
-            }
-            JexlNode sourceNode = getSourceNode(node);
-            if (contains(sourceNode)) {
-                if (log.isTraceEnabled()) {
-                    log.trace("Source of delayed node already exists in node set.");
-                }
-                if (remove(sourceNode)) {
-                    if (log.isTraceEnabled()) {
-                        log.trace("Removing source node " + sourceNode + " from node set prior to adding delayed predicate: " + node);
-                    }
+        String nodeKey = buildKey(node);
+        if (nodeMap.containsKey(nodeKey)) {
+            
+            // If the node mapped to our node key is delayed, do not overwrite.
+            if (isDelayed(nodeMap.get(nodeKey))) {
+                return false;
+            } else {
+                
+                // If we are not delayed but a delayed version of our node already exists, do not add.
+                if (!isDelayed(node)) {
+                    return false;
                 }
             }
```

Comment:
This could be simplified to a single if statement.

Reply:
Not sure why this is considered 'outdated'...

Reply:
Can do. Weird that it shows as outdated...it's definitely not..